dask/tests/test_optimization.py

```diff
@@ -1118,6 +1118,7 @@ def test_SubgraphCallable():
     f = SubgraphCallable(dsk, "h", ["in1", "in2"], name="test")
     assert f.name == "test"
     assert repr(f) == "test"
+    assert hash(f) == hash((tuple(dsk.keys()), "h", tuple(["in1", "in2"]), "test"))
```

Comment:
We don't actually care what the hash value is (what this test checks), we care that the hash is repeatable (hashing the same obj twice gives the same value) and works with `__eq__` to make dict lookups work. It would be good to test:

- That two instances of a `SubgraphCallable` with the same args hash the same and are equal
- That two different `SubgraphCallable` objects hash differently, and don't compare equal
- That `SubgraphCallable` objects can be keys in dicts. This would test both `__eq__` and `__hash__`.

Reply:
@jcrist, Thanks for the feedback! I removed the `self.dsk` attribute from the hash and accounted for the above scenarios in the test as removing the hash value check.